elytron/src/main/java/org/wildfly/extension/elytron/PolicyParser.java

```diff
@@ -82,21 +82,26 @@ void readPolicy(ModelNode parentAddress, XMLExtendedStreamReader reader, List<Mo
             }
         }
 
-        boolean providerFound = defaultPolicy == null;
-
-        while (reader.hasNext() && reader.nextTag() != END_ELEMENT) {
-            verifyNamespace(reader);
-            String localName = reader.getLocalName();
-            switch (localName) {
-                // Permission Mapper
-                case JACC_POLICY:
-                    providerFound = defaultPolicy.equals(parseJaccPolicy(addPolicy, reader, operations)) || providerFound;
-                    break;
-                case CUSTOM_POLICY:
-                    providerFound = defaultPolicy.equals(parseCustomPolicy(addPolicy, reader, operations)) || providerFound;
-                    break;
-                default:
-                    throw unexpectedElement(reader);
+        boolean providerFound = false;
+
+        if (defaultPolicy == null)
+        {
+            providerFound = true;
+        } else {
```

Comment:
This change doesn't feel quite correct - it skips the rest of the XML parsing are we sure that it should be skipped?